### PR TITLE
Feature/fear meter

### PR DESCRIPTION
**Pull request by hassanmohamed2003**
## Description
Implemented scare math

## Setting up testing environment
1. In the project window navigate to Project/Assets/Scenes and open "Game 1" scene.
2. Press Play.

#### Testing mechanics
Criteria:
- [x] when object is visible and/or hearable the NPC gets scared
- [x] NPC gets less and less scared and after 4 times it doesn't get scared anymore
- [x] Cooldown period between scaring works

## Checklist
_These checks should always be true for your pull request_
- [x] My pull request is for one story/feature.
- [x] Every individual commit in this pull request is logical.
- [x] All code, documentation, commits and player seen texts are in English.
- [x] I have deleted unused / unnecessary code.
- [x] If my edits need a change in documentation, then I have updated the documentation.
- [x] All code is in correspondence with the code conventions.fixed issues with possessable objects going through the map

### CODE REVIEW

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 1]
```diff
@@ -79,9 +100,59 @@ private void DetectTargets()
 
     public void OnNotify(ObservableObject observableObject)
     {
+        bool Hearing = observableObject.IsAudible;
```
Comment (dcsmit) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs]: Change this to audible

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 2]
```diff
@@ -79,9 +100,59 @@ private void DetectTargets()
...
+        bool Seeing = observableObject.IsVisible;
```
Comment (dcsmit) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs]: Change this to visible

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 3]
```diff
@@ -24,12 +25,32 @@ public class NpcSenses : MonoBehaviour, IObserver
     [Header("Reaction Settings")]
     [Tooltip("The delay between the NPC detecting a target and reacting to it."), Range(0f, 5f), SerializeField]
     private float _reactionDelay = 1f;
+    [Tooltip("Amount of time cooldown applies to NPC scare"), Range(0f, 10f), SerializeField]
+    private float _scaredCooldown = 2f;
+    [Header("Multiplier")]
+    [Tooltip("Multiplier to scare value when NPC hears ghost."), Range(0f, 5f), SerializeField]
+    private float _hearingMultiplier = 1f;
```
Comment (dcsmit) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs]: Change this to audible

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 4]
```diff
@@ -24,12 +25,32 @@ public class NpcSenses : MonoBehaviour, IObserver
...
+    [Tooltip("Multiplier to scare value when NPC sees ghost."), Range(0f, 5f), SerializeField]
+    private float _seeingMultiplier = 1f;
```
Comment (dcsmit) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs]: Change this to visible

**Review by dcsmit — CHANGES_REQUESTED** [review]
Works as it should, however, you should change a couple things

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 5]
```diff
@@ -79,9 +100,59 @@ private void DetectTargets()
 
     public void OnNotify(ObservableObject observableObject)
     {
+        bool hearable = observableObject.IsAudible;
```
Comment (SirCodeye) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs:103]: This should be audible

File: Assets/Scripts/NPC/Senses/NpcSenses.cs [hunk 6]
```diff
@@ -79,9 +100,59 @@ private void DetectTargets()
...
+        bool visable = observableObject.IsVisible;
```
Comment (dcsmit) [inline Assets/Scripts/NPC/Senses/NpcSenses.cs]: This is misspelled, it should be visible

**Review by dcsmit — CHANGES_REQUESTED** [review]

**Review by dcsmit — APPROVED** [review]
Looks good!

**Pull request merged**